tests/test_xarrayfits.py

```diff
@@ -150,20 +99,133 @@ def beam_cube(tmp_path_factory):
+
+    filenames = []
+
+    for i in range(3):
+        filename = str(path / f"data-{i}.fits")
+        filenames.append(filename)
+        primary_hdu = fits.PrimaryHDU(data)
+        primary_hdu.writeto(filename, overwrite=True)
+
+    return filenames
+
+
+def multiple_dataset_tester(datasets):
+    assert len(datasets) == 3
+
+    for xds in datasets:
+        expected = np.arange(np.prod(xds.hdu.shape), dtype=np.float64)
+        expected = expected.reshape(xds.hdu.shape)
+        assert_array_equal(xds.hdu.data, expected)
+
+    combined = xarray.concat(datasets, dim="hdu-0")
+    assert_array_equal(combined.hdu.data, np.concatenate([expected] * 3, axis=0))
+    assert combined.hdu.dims == ("hdu-0", "hdu-1")
+
+    combined = xarray.concat(datasets, dim="hdu-1")
+    assert_array_equal(combined.hdu.data, np.concatenate([expected] * 3, axis=1))
+    assert combined.hdu.dims == ("hdu-0", "hdu-1")
+
+    tds = [ds.expand_dims(dim="time", axis=0) for ds in datasets]
+    combined = xarray.concat(tds, dim="time")
+    assert_array_equal(combined.hdu.data, np.stack([expected] * 3, axis=0))
+    assert combined.hdu.dims == ("time", "hdu-0", "hdu-1")
+
+
+def test_list_files(multiple_files):
+    datasets = xds_from_fits(multiple_files)
+    return multiple_dataset_tester(datasets)
+
+
+def test_globbing(multiple_files):
+    path, _ = os.path.split(multiple_files[0])
+    datasets = xds_from_fits(f"{path}{os.sep}data*.fits")
+    return multiple_dataset_tester(datasets)
+
+
+def test_multiple_unnamed_hdus(multiple_hdu_file):
+    """Test hdu requests with hdu indexes"""
+    (ds,) = xds_from_fits(multiple_hdu_file, hdus=0)
+    assert len(ds.data_vars) == 1
+    assert ds.hdu.shape == (10, 10)
+    assert ds.hdu.dims == ("X", "Y")
+
+    (ds,) = xds_from_fits(multiple_hdu_file, hdus=[0, 2])
+    assert len(ds.data_vars) == 2
+
+    assert ds.hdu0.shape == (10, 10)
+    assert ds.hdu0.dims == ("hdu0-X", "hdu0-Y")
+
+    assert ds.hdu2.shape == (30, 40, 50)
+    assert ds.hdu2.dims == ("hdu2-X", "hdu2-Y", "hdu2-FREQ")
+
+
+def test_multiple_named_hdus(multiple_hdu_file):
+    """Test hdu requests with named hdus"""
+    (ds,) = xds_from_fits(multiple_hdu_file, hdus={0: "beam"})
+    assert ds.beam.dims == ("X", "Y")
+    assert ds.beam.shape == (10, 10)
+
+    (ds,) = xds_from_fits(multiple_hdu_file, hdus=["beam"])
+    assert ds.beam.dims == ("X", "Y")
+    assert ds.beam.shape == (10, 10)
+
+    (ds,) = xds_from_fits(multiple_hdu_file, hdus={0: "beam", 2: "3C147"})
+    assert ds.beam.dims == ("beam-X", "beam-Y")
+    assert ds.beam.shape == (10, 10)
+    assert ds["3C147"].dims == ("3C147-X", "3C147-Y", "3C147-FREQ")
+    assert ds["3C147"].shape == (30, 40, 50)
+
+    (ds,) = xds_from_fits(multiple_hdu_file, hdus=["beam", "3C147"])
+    assert ds.beam.dims == ("beam-X", "beam-Y")
+    assert ds.beam.shape == (10, 10)
+    assert ds["3C147"].dims == ("3C147-X", "3C147-Y", "3C147-FREQ")
+    assert ds["3C147"].shape == (10, 20, 30)
```

Comment:
@o-smirnov @landmanbester Sorry, I didn't highlight everything in my previous comment. This illustrates the naming schemas for hdus and dimensions.